File: api/src/endpoint/migration/list_github_installations.rs
use crate::{endpoint::Endpoint, resource::migration::GitHubInstallationResource};

pub struct ListGitHubInstallations;

impl Endpoint for ListGitHubInstallations {
    const PATH: &'static str = "/migration/github/installations";
    const METHOD: http::Method = http::Method::GET;

    type Request = ();
    type Response = ListGitHubInstallationsResponse;
}

pub type ListGitHubInstallationsResponse = Vec<GitHubInstallationResource>;

File: gitdot-web/packages/api/src/endpoint/migration/github/list-github-installations.ts
import { z } from "zod";
import { GitHubInstallationResource } from "../../../resource";
import type { Endpoint } from "../../endpoint";

export const ListGitHubInstallationsRequest = z.object({});
export type ListGitHubInstallationsRequest = z.infer<
  typeof ListGitHubInstallationsRequest
>;

export const ListGitHubInstallationsResponse = z.array(
  GitHubInstallationResource,
);
export type ListGitHubInstallationsResponse = z.infer<
  typeof ListGitHubInstallationsResponse
>;

export const ListGitHubInstallations = {
  path: "/migration/github/installations",
  method: "GET",
  request: ListGitHubInstallationsRequest,
  response: ListGitHubInstallationsResponse,
} as const satisfies Endpoint;
export type ListGitHubInstallations = typeof ListGitHubInstallations;
